src/linker/Linker/AssemblyResolver.cs

```diff
@@ -68,12 +72,57 @@ public AssemblyResolver (Dictionary<string, AssemblyDefinition> assembly_cache)
 			set { _context = value; }
 		}
 
+#if !FEATURE_ILLINK
+		// The base class's definition of GetAssembly is visible when using DirectoryAssemblyResolver.
+		AssemblyDefinition GetAssembly (string file, ReaderParameters parameters)
+		{
+			if (parameters.AssemblyResolver == null)
+				parameters.AssemblyResolver = this;
+
+			return ModuleDefinition.ReadModule (file, parameters).Assembly;
+		}
+#endif
+
+		private AssemblyDefinition ResolveWithPaths (AssemblyNameReference name, ReaderParameters parameters)
+		{
+			// Validate arguments, similarly to how the base class does it.
```

Comment:
This whole body should be better in Resolve method

Reply:
I've moved it into Resolve.